os/fs/smartfs/smartfs_utils.c

```diff
@@ -1297,12 +1294,12 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 		   Chainheader is written along with the 1st entry.
 		   So currently, smart_readsector will fail to invalidate CRC on empty sector
 		 */
-		tmp_buf = (char *)kmm_malloc(entrysize + sizeof(struct smartfs_entry_header_s));
+		tmp_buf = (char *)kmm_malloc(entrysize + sizeof(struct smartfs_chain_header_s));
```

Comment:
There was bug! very good..

Reply:
Yes but probably should be done in a separate commit as suggested by @thapav in previous comments.